Tasks/Gradle/sonarqube-common/sonarqube-common.ts

```diff
@@ -205,33 +223,37 @@ function getOrCreateSonarQubeStagingDirectory(): string {
     return sqStagingDir;
 }
 
-// Creates a string containing Markdown of a link to the SonarQube dashboard for this project.
-function createLinkToSonarQubeDashboard(sqBuildFolder: string): string {
-    var taskReport: TaskReport = getSonarQubeTaskReport(sqBuildFolder);
-    if (!taskReport) {
-        throw new Error(tl.loc('sqAnalysis_TaskReportInvalid'));
+/**
+ * Creates a string containing Markdown of the SonarQube quality gate status for this project.
+ * @param sqRunSettings SonarQubeRunSettings object for the applicable run
+ * @returns {string}    A Markdown-formatted report on the SonarQube quality gate status
+ */
+function createQualityGateStatusSection(sqReportBuilder: SonarQubeReportBuilder): Q.Promise<string> {
```

Comment:
The intent is not clear. Have a different method that describes the hook we are using for test. Then chain the fetch method to it.